src/python/grpcio_tests/tests/fork/methods.py

```diff
@@ -130,7 +134,9 @@ def start(self):
         self._process.start()
 
     def finish(self):
-        self._process.join()
+        self._process.join(timeout=_CHILD_FINISH_TIMEOUT_S)
+        if self._process.is_alive():
+            raise ValueError('Child process did not terminate')
```

Comment:
[`ValueError` is meant for invalid input.](https://docs.python.org/2/library/exceptions.html#exceptions.ValueError) I'd recommend subclassing [`RuntimeError`](https://docs.python.org/2/library/exceptions.html#exceptions.RuntimeError).